scripts/windows_install_packages_via_vcpkg.ps1

```diff
@@ -0,0 +1,87 @@
+# Install packages via vcpkg
+
+param (
+  [switch]$GenerateCacheTask = $false
+)
+
+# param() should be the first thing in the script file,
+# so ErrorActionPreference is defined here
+$ErrorActionPreference = "Stop"
+
+Function InstallVcpkgPackages()
+{
+  param (
+    [string]$VCPKG_PATH,
+    [string]$VCPKG_CACHE
+  )
+
+  # If the VCPKG_PATH or VCPKG_CACHE is not provided, try to get it from the environment
+  if (! ($VCPKG_PATH))
+  {
+    $VCPKG_PATH = ${Env:VCPKG_PATH}
+  }
+  if (! ($VCPKG_CACHE))
+  {
+    $VCPKG_CACHE = ${Env:VCPKG_CACHE}
+  }
+
+  $VCPKG_PACKAGES = @('gettext', 'icu', 'libxml2', 'libxslt', 'lz4', 'openssl', 'pkgconf', 'tcl', 'zlib', 'zstd')
+
+  cd ${VCPKG_PATH}
+  .\bootstrap-vcpkg.bat -disableMetrics
+  .\vcpkg.exe install --debug --clean-after-build --binarysource=files,${VCPKG_CACHE},readwrite --triplet=x64-windows $VCPKG_PACKAGES
+  if (!$?) { throw 'cmdfail' };
+}
+
+Function InstallAndPrepareImage()
+{
+  $VCPKG_PATH = "c:\vcpkg"
+  $VCPKG_CACHE_PATH = "${VCPKG_PATH}\binary_cache\"
+  $CIRRUS_BUILD_ID = ${Env:CIRRUS_BUILD_ID}
+
+  vcvarsall.bat x64
+  git clone --depth 1 https://github.com/Microsoft/vcpkg.git "${VCPKG_PATH}"
+  cd "${VCPKG_PATH}"
+
+  mkdir -p binary_cache, downloads
+  $ARTIFACT_NAME = "vcpkg_cache.zip"
+  $ARTIFACT_URL = "https://api.cirrus-ci.com/v1/artifact/build/${CIRRUS_BUILD_ID}/build-vcpkg-cache/vcpkg_cache_zip/${ARTIFACT_NAME}"
+
+  echo "Downloading  ${ARTIFACT_URL}"
+  curl.exe -fsSLO ${ARTIFACT_URL}
+  if (!$?) { throw 'cmdfail' };
+
+  echo "Extracting the cache"
+  7z.exe x ${ARTIFACT_NAME} -o"${VCPKG_CACHE_PATH}"
+  if (!$?) { throw 'cmdfail' };
+
+  echo "Installing packages via vcpkg"
+  InstallVcpkgPackages $VCPKG_PATH $VCPKG_CACHE_PATH
+
+  dir
+  (du -sh *) -or $true
+  (find . -name '*.exe' -or -name '*.pc') -or $true
+
+  $VCPKG_PKG_PREFIX = "${VCPKG_PATH}\installed\x64-windows"
+
+  $PKG_PATHS = "${VCPKG_PKG_PREFIX}\debug\lib;" +
+    "${VCPKG_PKG_PREFIX}\debug\bin;" +
+    "${VCPKG_PKG_PREFIX}\tools\pkgconf;"
+
+  [Environment]::SetEnvironmentVariable('PATH', ${PKG_PATHS} + [Environment]::GetEnvironmentVariable('PATH', 'Machine'), 'Machine')
+  [Environment]::SetEnvironmentVariable('PKG_CONFIG', 'pkgconf', 'Machine')
+}
```

Comment:
I wonder if we should set some other environment variable and change the pg test task to set them. That way we're not breaking existing tests.

E.g. something like PG_DEPS_PATH, PG_DEPS_PKG_CONFIG_PATHS? Then it'd just be two short statements to set them in the tests...